{"events":[{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","event":"call","id":175393,"lineno":72,"method_id":"handleDatabaseIssue_cannotCreateTx","path":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","object_id":1677603743,"value":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues@63fe339f"},"static":false,"thread_id":1},{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandler","event":"call","id":175394,"lineno":40,"method_id":"handleDatabaseConnectionIssue","parameters":[{"class":"org.springframework.transaction.CannotCreateTransactionException","kind":"req","name":"ex","object_id":533123465,"value":"org.springframework.transaction.CannotCreateTransactionException: db down"}],"path":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","receiver":{"class":"com.example.accesscontrol.exception.GlobalExceptionHandler","object_id":1084735306,"value":"com.example.accesscontrol.exception.GlobalExceptionHandler@40a7bf4a"},"static":false,"thread_id":1},{"elapsed":0.0,"event":"return","id":175395,"parent_id":175394,"return_value":{"class":"org.springframework.http.ResponseEntity","object_id":1570824593,"value":"<503 SERVICE_UNAVAILABLE Service Unavailable,{status=503, error=Database connection failed. Please try again later., timestamp=2025-08-14T15:53:54.210620100},[]>"},"thread_id":1},{"elapsed":3.0243E-4,"event":"return","id":175396,"parent_id":175393,"thread_id":1}],"classMap":[{"children":[{"children":[{"children":[{"children":[{"children":[{"labels":[],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:72","name":"handleDatabaseIssue_cannotCreateTx","static":false,"type":"function"}],"location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java","name":"GlobalExceptionHandlerTest$DatabaseIssues","static":false,"type":"class"},{"children":[{"labels":[],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java:40","name":"handleDatabaseConnectionIssue","static":false,"type":"function"}],"location":"src/main/java/com/example/accesscontrol/exception/GlobalExceptionHandler.java","name":"GlobalExceptionHandler","static":false,"type":"class"}],"name":"exception","type":"package"}],"name":"accesscontrol","type":"package"}],"name":"example","type":"package"}],"name":"com","type":"package"}],"version":"1.2","metadata":{"name":"Global exception handler $ database issues handle database issue _ cannot create tx","app":"accesscontrol","language":{"name":"java","version":"17.0.15+6-LTS","engine":"OpenJDK 64-Bit Server VM"},"client":{"name":"appmap-java","url":"https://github.com/appland/appmap-java"},"recorder":{"name":"junit","type":"tests"},"recording":{"defined_class":"com.example.accesscontrol.exception.GlobalExceptionHandlerTest$DatabaseIssues","method_id":"handleDatabaseIssue_cannotCreateTx"},"git":{"repository":"https://github.com/AlhaderMajd/accessControl-majd.git","branch":"main","commit":"7b2137103eb36e4ff706c6dac665a127d3b43f51"},"source_location":"src/test/java/com/example/accesscontrol/exception/GlobalExceptionHandlerTest.java:72","frameworks":[{"name":"JUnit","version":"5"}],"test_status":"succeeded"},"eventUpdates":{}}